opendbc/car/hyundai/interface.py

```diff
@@ -18,8 +18,11 @@ class CarInterface(CarInterfaceBase):
   def _get_params(ret: structs.CarParams, candidate, fingerprint, car_fw, experimental_long, docs) -> structs.CarParams:
     ret.carName = "hyundai"
 
-    hda2 = Ecu.adas in [fw.ecu for fw in car_fw]
-    CAN = CanBus(None, hda2, fingerprint)
+    # Check the CAM bus fingerprint for known HDA2 steering messages (0x50 or 0x110) to determine if it is an HDA2 car.
+    cam_can = CanBus(None, fingerprint).CAM
+    hda2 = 0x50 in fingerprint[cam_can] or 0x110 in fingerprint[cam_can]
+
+    CAN = CanBus(None, fingerprint, hda2)
```

Comment:
Code is now self-explanatory and groups together logically.

```suggestion
    cam_can = CanBus(None, fingerprint).CAM
    hda2 = 0x50 in fingerprint[cam_can] or 0x110 in fingerprint[cam_can]
    CAN = CanBus(None, fingerprint, hda2)
```